app/web/src/store/realtime/heimdall.ts

```diff
@@ -61,39 +125,41 @@ export const init = async (
 ) => {
   if (!ranInit.value) {
     // eslint-disable-next-line no-console
-    console.log("🌈 initializing bifrost...");
-    const start = performance.now();
+    console.log("🌈 calling init...");
     await db.setBearer(workspaceId, bearerToken);
 
     const { port1, port2 } = new MessageChannel();
     // This message fires when the lock has been acquired for this tab
     port1.onmessage = () => {
       db.setRemote(tabDbId);
+      // eslint-disable-next-line no-console
+      console.log("🌈 lock acquired by this tab");
       lockAcquired.value = true;
-      lockAcquiredBroadcastChannel.postMessage("acquired");
+      lockAcquiredBroadcastChannel.postMessage(tabDbId);
     };
 
-    // We are deliberately not awaiting this promise, since it blocks forever on the tabs that do not get the lock
+    // We are deliberately not awaiting this promise, since it blocks forever on
+    // the tabs that do not get the lock
     tabDb.initBifrost(Comlink.proxy(port2));
 
-    const end = performance.now();
     ranInit.value = true;
     queryClient = _queryClient;
-    // eslint-disable-next-line no-console
-    console.log(`...initialization completed [${end - start}ms] 🌈`);
   }
 
-  // If both tabs are refreshed at the same time, this can falsely indicate that a tab has the lock,
-  // but that tab has actually been refreshed just *after* this call, so *we* now have the lock.
-  // adding 2.5 second timeout here ensures that there is enough time for the lock to be resolved
-  // in a multitab scenario before we begin cold start. (This only matters if 2+ tabs are refreshed
-  // at more or less the same time, in the normal scenario we will indicate lock acquisition via
-  // the broadcast channel)
+  // If both tabs are refreshed at the same time, this can falsely indicate that
+  // a tab has the lock, but that tab has actually been refreshed just *after*
+  // this call, so *we* now have the lock.  adding 1.5 second timeout here
+  // ensures that there is enough time for the lock to be resolved in a multitab
+  // scenario before we begin cold start. (This only matters if 2+ tabs are
+  // refreshed at more or less the same time, in the normal scenario we will
+  // indicate lock acquisition via the broadcast channel)
   setTimeout(async () => {
-    if (await db.hasRemote()) {
+    if ((await db.hasRemote()) && !(await tabDb.hasDbLock())) {
```

Comment:
similarly here, if we `hasRemote` what does it matter that "its not my tabdb that has it"?